accounting/bank/models.py

```diff
@@ -95,33 +81,11 @@ class BankAccount(models.Model):
     currency = CurrencyField(
         help_text=_("The currency expected to be held in this bank account."))
     type = models.CharField(
-        max_length=30, choices=BANK_ACCOUNT_TYPES,
+        max_length=30, choices=BankAccountType.choices,
         help_text=_("Whether this is a checking or savings account."))
-    iban = IBANField(
-        blank=True, null=True,
-        help_text=_("The unique International Bank Account Number of the provider's bank account."))
-    bic = BICField(
-        blank=True, null=True,
-        help_text=_("The 11-character SWIFT / BIC (Business Identifier Code) code used to identify a "
-                    "bank or financial institution globally."))
-    abn = AUBusinessNumberField(
-        blank=True, null=True,
-        help_text=_("The 11-character Australian Business Number used to identify business entities in Australia."))
-    bsb = AUBankStateBranchField(
-        blank=True, null=True,
-        help_text=_("The 6-character Bank State Branch code used as a bank identifier in Australia."))
-    vat = VATINField(
-        blank=True, null=True,
-        help_text=_("The Value Added Tax identification number used in and required by some countries. "
-                    "This can also be used to store a General Sales Tax (GST) identification number."))
-    account_number = models.CharField(
-        max_length=30, blank=True, null=True,
-        help_text=_("The bank account number used to help identify the account. "
-                    "Required for only some countries."))
-    routing_number = models.CharField(
-        max_length=30, blank=True, null=True,
-        help_text=_("The bank account routing number used to help identify the account. "
-                    "Required for only some countries."))
+    identification = JSONField(
+        blank=True, default=IDENTIFICATION_SCHEMA,
+        help_text=_("The unique combination of identification information for this bank account."))
```

Comment:
I thought we had agreed to using a JSONField for this.

I'd have to make the TransferWise API client for this from the ground up, since no Python client even exists yet. That's not a non-trivial amount of work to test and get right (as usual with API integrations), so I did it this way for this iteration.

I plan on doing this next sprint.

Reply:
We had agreed on using a JSONField *if we needed to store the information on our side*. And for transfers supported by Transferwise, we don't, no? Again, I don't think you need to implement the transferwise integration for this, since Alicia will still create the batch payment manually there. But since that information is already there, for transferwise recipients I odn't think it makes sense to get her to copy all that information manually to our DB.

Also, the more I think about it, the less I think we need to automate the recipient creation in transferwise. They already have a nice interface for this, and this is only ever done once per developer - it's not like if we had dozen of new developers to register every month. It could be a perfectly sane workflow for now, when adding a new developer, to have the admin go create the recipient in Transferwise, and just enter the recipient id in the admin.

Reply:
:+1: yep your above comment cleared things up for me regarding bank details finally. The account number + BIC/SWIFT, and the recipient ID, should cover all cases, I hope.